File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int start = 0;
        double res = Integer.MIN_VALUE;
        double cur = 0;
        for (int end = 0; end < nums.length; end++) {
            cur += nums[end];
            if (end - start + 1 == k) {
                res = Math.max(cur, res);
                cur -= nums[start];
                start++;
            }
        }
        return res/k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        res = float('-inf')
        cur = 0
        start = 0
        for end in range(len(nums)):
            cur += nums[end]
            if end - start + 1 == k:
                res = max(res, cur)
                cur -= nums[start]
                start += 1
        return res / k